docs/guides/dependency-learning-resources.md

```diff
@@ -0,0 +1,95 @@
+# Resources for learning about our dependencies
+
+Before reading anything else in this guide you should read this article: https://medium.freecodecamp.org/how-to-escape-tutorial-purgatory-as-a-new-developer-or-at-any-time-in-your-career-e3a4b2384a40
+
+The main point is that reading tutorials and documentation isn't worth it if you don't also get to use it. This article is therefore meant not as a study guide you should go through step by step and complete all the resources listed here, but rather you should use it to maybe skim the most important parts, and when you are working on an issue that requires more knowledge than you currently have about a technology used for that issue, you can consult this article and find good resources to learn more about this.
+
+With that said, here are some resources listed by technology, feel free to add more resources to this article if you find any useful ones as you look around the internet. Also remember that we don't always use the latest version of each of these libraries, so you may find features we don't have if you don't look back into the versioning of the documentation and find the relevant one. Also remember that if you're ever stuck with how to do something with a library, often Google and StackOverflow have the answer! There's a great community of programmers out there!
+
+## Javascript
+
+We use a lot of new Javascript, and Javascript is also very different from other languages since it is very asynchronous in nature. We really recommend watching this talk: https://www.youtube.com/watch?v=8aGhZQkoFbQ&t=8s to understand how the event loop works. Here are some more resources on promises and asynchronous functions which we use heavily:
+
+- [an extensive tutorial on Promises](https://developers.google.com/web/fundamentals/primers/promises)
+- [a nice article on asynchronous functions from the perspective of Promises](https://medium.com/@bluepnume/learn-about-promises-before-you-start-using-async-await-eb148164a9c8)
+- [an async/await tutorial](https://codeburst.io/javascript-es-2017-learn-async-await-by-example-48acc58bad65)
+- [a good article on bad habits in async/await](https://medium.freecodecamp.org/avoiding-the-async-await-hell-c77a0fb71c4c)
+- [an async await tutorial](https://hackernoon.com/6-reasons-why-javascripts-async-await-blows-promises-away-tutorial-c7ec10518dd9)
+
```

Comment:
Once done understanding Promises, I think an article or two + just looking at code examples is good enough to understand async/await. So here might be an awesome place to link couple of great examples of async/await code.